src/main/java/seedu/address/model/person/Phone.java

```diff
@@ -36,14 +35,26 @@ public static boolean isValidPhone(String test) {
 
     @Override
     public String toString() {
+        if (value == null) {
+            return "No phone number";
+        }
         return value;
     }
 
     @Override
     public boolean equals(Object other) {
-        return other == this // short circuit if same object
-                || (other instanceof Phone // instanceof handles nulls
-                && value.equals(((Phone) other).value)); // state check
+        if (other == this) {
+            // short circuit if same object
+            return true;
+        } else if (other instanceof Phone) {
+            // instanceof handles nulls
+            if (value == null || ((Phone) other).value == null) {
+                // at least 1 of the values are null
+                return value == ((Phone) other).value;
+            }
+            return value.equals(((Phone) other).value); // state check
```

Comment:
Similarly, can use `Objects.equals`.

Reply:
Will cause NullPointerException if I did that

Reply:
Will cause NullPointerException if I did that